pandas/core/arrays/string_arrow.py

```diff
@@ -0,0 +1,623 @@
+from __future__ import annotations
+
+from distutils.version import LooseVersion
+from typing import TYPE_CHECKING, Any, Sequence, Type, Union
+
+import numpy as np
+
+from pandas._libs import lib, missing as libmissing
+from pandas.util._validators import validate_fillna_kwargs
+
+from pandas.core.dtypes.base import ExtensionDtype
+from pandas.core.dtypes.dtypes import register_extension_dtype
+from pandas.core.dtypes.missing import isna
+
+from pandas.api.types import (
+    is_array_like,
+    is_bool_dtype,
+    is_integer,
+    is_integer_dtype,
+    is_scalar,
+)
+from pandas.core.arraylike import OpsMixin
+from pandas.core.arrays.base import ExtensionArray
+from pandas.core.indexers import check_array_indexer, validate_indices
+from pandas.core.missing import get_fill_func
+
+try:
+    import pyarrow as pa
+except ImportError:
+    pa = None
+else:
+    try:
+        import pyarrow.compute as pc
+    except ImportError:
+        pass
```

Comment:
when would this be true? (e.g. have pyarrow but now pyarrow.compute)? can you comment on min version

Reply:
yes there is no pyarrow.compute in our min supported version, 0.15.1.

for testing/dev i've only setup environments with no pyarrow, 0.15.1, 1.0.0 and 2.0.0

we have added `_chk_pyarrow_available` here that checks `if pa is None or LooseVersion(pa.__version__) < "1.0.0":` and raises on construction. so that's why pa and pc are imported independently.

I'm not sure what the min pyarrow version is that 'could' be supported is. @jorisvandenbossche? or whether it's actually relevant.

Reply:
`pyarrow=1.0` should be sufficient for the basic usage. With 2.0+ you will get a lot more string algorithms. In `fletcher` I have started using `hasattr(pc, '<algorithm>')` for that in the `str` accessor overloads.

Reply:
Functionality is the same, performance will be different.

Reply:
ok, to be honest ok with requiring >=1.0 for this change its fine for now can revist.

Reply:
Anaconda doesn't even yet provide pyarrow 1.0 (it's only on conda-forge), so let's leave for later / separate discussion. 

> I'm not sure what the min pyarrow version is that 'could' be supported is.

1.0 is the minimum we need for decent support here in the PR (and once we add the string methods, we will also need to have conditionals on the actual pyarrow version (1.0, 2.0, 3.0), as the more recent have more string compute algos available. But so that is for a next PR.

Reply:
> Anaconda doesn't even yet provide pyarrow 1.0 (it's only on conda-forge), so let's leave for later / separate discussion.

interesting